source/validate_id.cpp

```diff
@@ -417,7 +417,7 @@ bool idUsage::isValid<SpvOpTypeArray>(const spv_instruction_t* inst,
   } else if (5 == constInst->words.size()) {
     uint64_t value =
         constInst->words[3] | ((uint64_t)constInst->words[4]) << 32;
-    bool signedness = constResultType->second.inst->words[3];
+    bool signedness = constResultType->second.inst->words[3] ? true : false;
```

Comment:
Yeah, I wish warnings cross-compiler were more consistent. Or there was a good way to get the superset of all warnings without playing whack-a-mole. Or that the C++ spec was well-defined enough such that all compilers would have the same spec-defined warnings (especially in cases like this, where you aren't even doing anything crazy!).

I think the readability argument is less about the conversion and more about when it happens.
```
if (words[3]) {  // clear to reader bool conversion is happening
bool foo = words[3] ? true : false;  // also clear bool conversion is happening
bool foo = words[3];  // not clear, requires understanding that words[3] is uint32_t but the semantics of the particular index 3 is boolean
```

When we were migrating to C++11 a coworker once came up with a pretty good rule of thumb: if switching your lvalue type to 'auto' requires an unfamiliar reader to leave the file to figure out what the type is your naming is unclear or you're missing a cast.

But, like anything readability-related: totally subjective :P